tcms/kiwi_auth/tests/test_profiles.py

```diff
@@ -35,26 +28,20 @@ def test_user_can_view_their_own_profile(self):
         self.assertContains(response, 'name="_save"')
 
     def test_user_cant_view_profile_of_another_user_without_permission(self):
-        logged_in = self.client.login(  # nosec:B106:hardcoded_password_funcarg
-            username=self.tester.username,
-            password='password')
-        self.assertTrue(logged_in)
-
         url = reverse('tcms-profile', args=[self.somebody_else.username])
         response = self.client.get(url, follow=True)
         self.assertIsInstance(response, HttpResponseForbidden)
 
     def test_user_can_view_profile_of_another_user_with_permission(self):
         user_should_have_perm(self.tester, 'auth.view_user')
 
-        logged_in = self.client.login(  # nosec:B106:hardcoded_password_funcarg
-            username=self.tester.username,
-            password='password')
-        self.assertTrue(logged_in)
-
         url = reverse('tcms-profile', args=[self.somebody_else.username])
         response = self.client.get(url, follow=True)
 
         self.assertContains(response, self.somebody_else.username)
         self.assertContains(response, self.somebody_else.email)
         self.assertNotContains(response, 'name="_save"')
+
+    def test_view_if_user_is_invalid(self):
+        response = self.client.get(reverse("tcms-profile", args=["kiwi-tester"]))
```

Comment:
"kiwi-tester" sounds like something that somebody may use in the future. Better use a more verbose username like "non-existing-username" or similar.